librz/include/rz_il/rzil_opcodes.h

```diff
@@ -296,8 +296,8 @@ struct rzil_op_bool_inv_t {
  *  load m k is the value associated with the key k in the memory m.
  */
 struct rzil_op_load_t {
-	int mem; ///< index of the memory in VM (different from the temp_val_list)
-	RzILOp *key; ///< index of the RzBitVector key (address)
+	RzILOp *key; ///< memory index of the RzBitVector key (address)
+	ut32 len; ///< n of bits to read
 };
```

Comment:
What you have implemented here is not `load` and `store` (which load single "bytes"), but `loadw` and `storew` which load entire words of a given length.
See
http://binaryanalysisplatform.github.io/bap/api/master/bap-core-theory/Bap_core_theory/Theory/module-type-Basic/index.html#val-load
http://binaryanalysisplatform.github.io/bap/api/master/bap-core-theory/Bap_core_theory/Theory/module-type-Basic/index.html#val-loadw

We should do this in the same way. Essentially that would mean leaving the original implementation of load/store and adding what you have written as loadw/storew.